src/deepsparse/transformers/pipelines/text_generation.py

```diff
@@ -0,0 +1,359 @@
+
+from deepsparse import Pipeline
+from deepsparse.transformers.engines import NLDecoderEngine
+from deepsparse.transformers.helpers import get_onnx_path_and_configs
+from deepsparse.transformers.pipelines import TransformersPipeline
+
+
+__all__ = ["TextGenerationPipeline"]
+
+
+class TextGenerationInput(BaseModel):
+    sequence: str = Field(
+        description="The input sequence to generate the text from.",
+    )
+    return_logits: bool = Field(
+        default=False,
+        description="A flag that indicates whether to return "
+        "the logits for the generated text sequence. ",
+    )
+
+
+class TextGenerationOutput(BaseModel):
+    sequence: str = Field(
+        description="The generated text sequence.",
+    )
+    logits: Optional[numpy.ndarray] = Field(
+        default=None,
+        description="The logits for the generated text sequence."
+        "The logits have dimensions "
+        "[batch_size, sequence_length, vocab_size]",
+    )
+
+    class Config:
+        arbitrary_types_allowed = True
+
+
+@Pipeline.register(
+    task="text_generation",
+    task_aliases=["codegen", "opt", "bloom"],
+)
+class TextGenerationPipeline(TransformersPipeline):
+    """
+    Pipeline for text generation tasks.
+
+    :param deterministic: if True, the pipeline will sample from
+        the probability distribution computed from the logits.
+        If False, the pipeline will get the next token by applying
+        an argmax function to the logits.
+    :param sampling_temperature: the temperature to use when sampling
+        from the probability distribution computed from the logits.
+        Higher values will result in more random samples.
+    :param max_generated_tokens: the maximum number of tokens to generate
+        given the input sequence. If None, the model will generate
+        tokens until the end of the sequence is reached.
+        Otherwise, it will generate up to the maximum number of tokens or end of
+        sequence is reached.
+    :param prompt_batch_threshold: the threshold for the ratio of running the prompt
+        as a single inference vs running the prompt auto-regressively.
+        If the number of input sequences divided by the max sequence length is
+        greater than the threshold, the prompt will be run as a single inference.
+        Default is None, which will always run auto-regressively.
+    :param force_max_tokens: if True, the pipeline will generate the maximum number
+        of tokens supplied even if the stop token is reached.
+    :param kwargs: kwargs to pass to the TransformersPipeline
+    """
+
+    def __init__(
+        self,
+        deterministic: bool = True,
+        sampling_temperature: float = 1.0,
+        max_generated_tokens: Optional[int] = 4096,
+        prompt_batch_threshold: Optional[float] = None,
+        force_max_tokens: bool = False,
+        **kwargs,
+    ):
+        if kwargs["engine_type"] == "deepsparse":
```

Comment:
remove it, add flag whether or not use the internal (DeepSparse cache), default to False if set to True raise an error (for now)